module/extension/Director/src/Director.hpp

```diff
@@ -373,20 +375,49 @@ namespace module::extension {
          */
         void run_task_pack(const TaskPack& pack);
 
+        /**
+         * A RunReasonLock instance will reset the current_run_reason back to OTHER_TRIGGER when it is destroyed.
+         *
+         * This is used so that the run reason can temporarily be set to something else, and then reset back to the
+         * default if there is an exception or other error.
+         */
+        using RunReasonLock = std::unique_ptr<void, std::function<void(void*)>>;
```

Comment:
A part of me wants you to replace this with a struct that is purpose built for this, but i dont know how big a part of me that is :stuck_out_tongue:

Reply:
A very small part